printer/kubectl_diff.go

```diff
@@ -0,0 +1,39 @@
+package printer
+
+import (
+	"bufio"
+	"fmt"
+	"github.com/kubecolor/kubecolor/config"
+	"io"
+	"strings"
+)
+
+type DiffPrinter struct {
+	Theme *config.Theme
+}
+
+func (p *DiffPrinter) Print(r io.Reader, w io.Writer) {
```

Comment:
It would be grand with some tests for this printer. We have a custom "integration tests" runner with a custom file format used to detect regressions.

To add tests, create a file `test/corpus/kubectl_diff.txt` with the content:

```
===
$ kubectl diff -f service.yaml
===

diff -u -N /tmp/LIVE-2333300441/v1.Service.default.foobar /tmp/MERGED-3286728941/v1.Service.default.foobar
--- /tmp/LIVE-2333300441/v1.Service.default.foobar      2024-11-24 15:37:23.524788761 +0100
+++ /tmp/MERGED-3286728941/v1.Service.default.foobar    2024-11-24 15:37:23.525788783 +0100
@@ -0,0 +1,28 @@
+apiVersion: v1
+kind: Service
+metadata:
+  creationTimestamp: "2024-11-24T14:37:23Z"
+  labels:
+    app: foobar
+  name: foobar
+  namespace: default
+  uid: cfb34072-a09c-44c0-956b-54db81a3c2cb
+spec:
+  clusterIP: 10.128.0.0
+  clusterIPs:
+  - 10.128.0.0
+  internalTrafficPolicy: Cluster
+  ipFamilies:
+  - IPv4
+  ipFamilyPolicy: SingleStack
+  ports:
+  - name: http
+    port: 80
+    protocol: TCP
+    targetPort: 80
+  selector:
+    app: foobar
+  sessionAffinity: None
+  type: ClusterIP
+status:
+  loadBalancer: {}

---
```

Followed by running the following make target which will update the test to include the "expected" output:

```bash
make corpus-update
```

This will help us in the future to see if any change will affect the `kubectl diff` output.

Feel free to add more tests than the one I provided, such as some tests showing some lines deleted as well.

Reply:
Yeah this will be done by me. :)

Reply:
I've fixed an issue in the corpus scanner. 

Kubectl diff creates a line, which starts with `---`. The scanner falsely detects this as end of test.

I've adjusted the logic, so that `scanUntils` searches for a full line of the same char (min 3 times the same char)

Finally, I've adjusted the tests to contain 3 tests (all plus, plus and minus, minus for a list attribute). Unfortunate, I did not find a real-world diff for all lines removed :/

Reply:
I think there isn't a use case for all lines removed. :)

Reply:
done!